app/controllers/transfers_controller.rb

```diff
@@ -6,12 +6,34 @@ def create
       transfer_params.merge(source: @source, destination: @account)
     )
 
-    begin
-      transfer.save!
-    rescue ActiveRecord::RecordInvalid
-      flash[:error] = transfer.errors.full_messages.to_sentence
-    end
+    transfer.save!
     redirect_to redirect_target
+  rescue ActiveRecord::RecordInvalid
+    flash[:error] = transfer.errors.full_messages.to_sentence
+  end
+
+  def new
+    transfer_factory = TransferFactory.new(
+      current_organization,
+      current_user,
+      params[:offer],
+      params[:destination_account_id]
```

Comment:
The term `destination_account` would be better off as [payee](https://www.collinsdictionary.com/dictionary/english/payee). Seems more appropriate to this domain.